### PR TITLE
test/helpers module to store AppendRandomString helper fn

### PR DESCRIPTION
**Pull request by odra**
Fixes: #226

I have added the helper function`AppendRandomString` into `test/helpers` module (helpers being a new module), sample usage:

```go
package pkg

import (
	"github.com/knative/pkg/test/helpers"
	"testing"
	"strings"
)


func TestAppendRandomString(t *testing.T) {
	s := helpers.AppendRandomString("myprefix")

	if !strings.HasPrefix(s, "myprefix") {
		t.Fatalf("Failed to verify prefix: %s", s)
	}

	if len(s) != 18 {
		t.Fatalf("Failed to verify string size: %d", len(s))
	}

	t.Logf("---%s---", s)
}
```

I was looking for a first good issue and this one looked simple enough :)

### CODE REVIEW

**Comment by odra** [comment]
I signed it!

File: test/helpers/data.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+func AppendRandomString(prefix string) string {
+	once.Do(initSeed)
+	suffix := make([]byte, randSuffixLen)
+
+	rndMutex.Lock()
+	for i := range suffix {
+		suffix[i] = letterBytes[r.Intn(len(letterBytes))]
+	}
+	rndMutex.Unlock()
```
Comment (markusthoemmes) [inline test/helpers/data.go]: Please `defer` this right after getting the Lock above. The minimal gain in concurrency is not worth the risk of a panic in between I think.

File: test/helpers/data.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	once     sync.Once
+)
+
+func initSeed() {
+	seed := time.Now().UTC().UnixNano()
+	r = rand.New(rand.NewSource(seed))
+	rndMutex = &sync.Mutex{}
+}
+
+func AppendRandomString(prefix string) string {
```
Comment (jessiezcc) [inline test/helpers/data.go]: the same function exists at https://github.com/knative/serving/blob/master/test/crd.go already. 
But it makes sense to move it out of serving to pkg. Will you pls remove it from serving?

**Comment by jessiezcc** [comment]
the same function exists at https://github.com/knative/serving/blob/master/test/crd.go already.
But it makes sense to move it out of serving to pkg. Added comments in code.

File: test/helpers/data.go (hunk 2 above)
Comment (odra) [reply test/helpers/data.go]: Sure, sorry I missed it

Comment (odra) [reply test/helpers/data.go]: Ah right, I was going to remove it right after the pr is in good shape :)

Comment (odra) [reply test/helpers/data.go]: Does it make sense to remove it from here as well? https://github.com/knative/build-pipeline/blob/master/test/randstring.go

File: test/helpers/data.go (hunk 1 above)
Comment (odra) [reply test/helpers/data.go]: Done

File: test/helpers/data.go (hunk 2 above)
Comment (jessiezcc) [reply test/helpers/data.go]: yes, all duplication

Comment (jessiezcc) [reply test/helpers/data.go]: can you pls open a git issue to track the next step?

Comment (odra) [reply test/helpers/data.go]: I've created one issue in this repository: https://github.com/knative/pkg/issues/274 and also two new ones on each repository to track the actual work:

https://github.com/knative/build-pipeline/issues/490
https://github.com/knative/serving/issues/3174

**Pull request merged**